hedera-node/src/main/java/com/hedera/services/txns/contract/operation/HederaCallCodeOperation.java

```diff
@@ -22,38 +22,32 @@
  *
  */
 
-import com.hedera.services.contracts.execution.SoliditySigsVerifier;
-import com.hedera.services.utils.EntityIdUtils;
+import org.hyperledger.besu.datatypes.Address;
 import org.hyperledger.besu.evm.EVM;
 import org.hyperledger.besu.evm.frame.MessageFrame;
 import org.hyperledger.besu.evm.gascalculator.GasCalculator;
 import org.hyperledger.besu.evm.operation.CallCodeOperation;
 
+import javax.inject.Inject;
 import java.util.Optional;
-import java.util.Set;
+import java.util.function.BiFunction;
 
 public class HederaCallCodeOperation extends CallCodeOperation {
-	private final SoliditySigsVerifier sigsVerifier;
+	final BiFunction<Address, MessageFrame, Boolean> addressValidator;
 
-	public HederaCallCodeOperation(SoliditySigsVerifier sigsVerifier,
-								   GasCalculator gasCalculator) {
+	@Inject
+	public HederaCallCodeOperation(GasCalculator gasCalculator,
+			BiFunction<Address, MessageFrame, Boolean> addressValidator) {
 		super(gasCalculator);
-		this.sigsVerifier = sigsVerifier;
+		this.addressValidator = addressValidator;
 	}
 
 	@Override
 	public OperationResult execute(MessageFrame frame, EVM evm) {
-		final var account = frame.getWorldUpdater().get(to(frame));
-		if (account == null) {
+		if (!addressValidator.apply(to(frame), frame)) {
 			return new OperationResult(
 					Optional.of(cost(frame)), Optional.of(HederaExceptionalHaltReason.INVALID_SOLIDITY_ADDRESS));
 		}
-		final var accountId = EntityIdUtils.accountParsedFromSolidityAddress(account.getAddress().toArray());
-		if (!sigsVerifier.allRequiredKeysAreActive(Set.of(accountId))) {
-			return new OperationResult(
-					Optional.of(cost(frame)), Optional.of(HederaExceptionalHaltReason.INVALID_SIGNATURE)
-			);
-		}
```

Comment:
Shouldn't we check for signature if `receiverSigRequred=true`?

Reply:
I need to pull from the current ethereumj-besu-swap branch.  The idea for non-precompile code is that this will be the same.